python/cudf/dataframe/dataframe.py

```diff
@@ -673,7 +673,11 @@ def set_index(self, index):
                 df[k] = self[k].set_index(index)
             return df
 
-    def reset_index(self):
+    def reset_index(self, drop=False):
+        if not drop:
+            name = self.index.name or 'index'
+            self[name] = self.index
+            self = self[[name] + list(self.columns[:-1])]
```

Comment:
This runs inplace instead of returning by shallow copy.

Reply:
So I should call `.copy()`?

Reply:
> So I should call `.copy()`?

I think your best bet would be to instantiate a new DataFrame object and then follow the logic you have here and it should be good to go.